### ISSUE TITLE
Board no longer displays ANY train times

### ISSUE THREAD

**Issue by condeepadunov** [metro-sign/dc-metro#4]
It still shows the headers (LN, DEST, and MIN) but no train times. Hasn't shown train times in at least a week, and I can't figure out why. If I had to guess, I'd guess that it's tied to a slight change in the way WMATA displays train arrivals. **I believe this feature illustrating how busy a train is expected to be is new, and it's screwing up how the board reads the API.** But I could be wrong. Anyone else have any thoughts?

<img width="387" alt="Screen Shot 2021-10-12 at 10 31 11 PM" src="https://user-images.githubusercontent.com/92415222/137057376-6cf6a73f-34c8-48ba-8ad1-93be41e04b0f.png">

<img width="377" alt="Screen Shot 2021-10-12 at 10 33 40 PM" src="https://user-images.githubusercontent.com/92415222/137057402-fea34d84-9af4-4bdc-8cf9-be7c70f880ea.png">

**Comment by condeepadunov** [comment]
Update: As of today, it is mysteriously resolved. Displaying arrivals once again.

**Comment by DawsonJung** [comment]
Mine started doing the exact same thing. I see LN DEST and MIN but no data populated underneath

**Comment by kevinhollerbach** [comment]
I'm having the same issue. It seems like the WMATA API is down. I recommend reaching out to the WMATA API team to let them know: api-support@wmata.com

**Comment by rmwalley** [comment]
I emailed someone at wmata:    Javier got back in touch with me and was able to use the API along with the GitHub repository for the board, and his API key.  He indicated that it is most likely something wrong with my local configuration and I should try to debug the code (which I have never done and don't know how to go about doing that.  I am trying some different versions of CircuitPy along with various Lib and src repositories to see if I can get a working combination.  Otherwise, plan on getting back in touch with Javier to see if he can help me out further.

**Comment by kevinhollerbach** [comment]
That's strange, I was getting "resource not found" errors when testing the WMATA API, but I'll keep testing and see if I find a solution. Thanks for reaching out.

**Comment by MantisToboggan5** [comment]
I am having the same issue on my board, I've tried regenerating and updating API key.  Looking at the reports on the dev site it looks like the API calls stopped being received from WMATA side Aug 26th.  Not sure if there is something changed from WMATA side then.

**Comment by rmwalley** [comment]
I agree--the API calls stopped at the end of August.  Will bring this up with Javier at the above listed email and see what he has to say.  Promise to report back any info I get.

**Comment by rmwalley** [comment]
Just a quick followup:  the contact at WMATA basically said he could not help and recommended that we hire someone to figure out the issues on our end.  Also, he seemed upset that I shared his WMATA contact info, so I removed it from my post above.  Any other ideas that you may have, please let me know.  Not sure what to do now other than maybe use my board for a new project

**Comment by kevinhollerbach** [comment]
Thanks. I have a couple of tests that I will test tonight or tomorrow and let you know if I find a solution. I think one of the methods for pulling data from the API has changed slightly, so I'm hoping this will solve it!

**Comment by fernat** [comment]
Not sure if this is the issue, but I was having a similar issue on a different board and this resolved:

https://forums.adafruit.com/viewtopic.php?t=214258

https://learn.adafruit.com/upgrading-esp32-firmware/upgrade-all-in-one-esp32-airlift-firmware

**Comment by kevinhollerbach** [comment]
Yes, this was the issue! I also found this last night after trying a number of different solutions. Once I updated the esp32 firmware, the board started working as expected. 

There was some issue with the wifi chip on the Matrix board connecting to the WMATA API HTTPS server – I saw an error suggesting the board was not providing an SSL certificate. There must have been something in the firmware update that allowed the matrix M4 to work better with SLS / HTTPS.

It's working!!

![Image](https://github.com/user-attachments/assets/34b36919-8b37-4e3e-9166-8e05026b84cb)

**Comment by rmwalley** [comment]
Did you have to upgrade the firmware and then reload everything else or did you just upgrade firmware and it just worked?

**Comment by kevinhollerbach** [comment]
If I remember correctly, installing the new esp32 firmware wipes the memory, so you need to reinstall everything after that.

**Comment by DawsonJung** [comment]
> Not sure if this is the issue, but I was having a similar issue on a different board and this resolved:
> 
> https://forums.adafruit.com/viewtopic.php?t=214258
> 
> https://learn.adafruit.com/upgrading-esp32-firmware/upgrade-all-in-one-esp32-airlift-firmware

This worked for me as well. Thank you!

![Image](https://github.com/user-attachments/assets/004030a4-dc9b-4cd7-b989-2f36d9bcccd4)

**Comment by rmwalley** [comment]
Thanks for the assistance!   This solution worked for me as well

![image](https://github.com/user-attachments/assets/932112ad-bbf5-44c0-80a5-ddc63a4d20a5)

**Comment by btr2tv** [comment]
Can someone help out a real idiot here? I'm getting stuck in the steps in the pic below here after I upload NINA. It successfully installs, but truly nothing happens when I hit the reset button or unplug and reinsert the usb. Is it supposed to mount like a USB? Or am I supposed to reinstall CircuitPython a different way?

<img width="959" height="738" alt="Image" src="https://github.com/user-attachments/assets/441d0043-bffe-4f0d-9a77-04d181d6bb89" />

If I reset the whole thing (double click reset) and reinstall CircuitPython (like the original instructions of this), everything from my previous is still there, but obviously isn't working given ESP32 firmware is now required. I did backup just in case so I hope I can just drag and drop once I figure out what I'm doing wrong.
